admin/cluster_setup.py

```diff
@@ -0,0 +1,477 @@
+                provider_config=provider_config,
+            )
+
+    def _provider_config_missing(self, provider):
+        """
+        :param str provider: The name of the missing provider.
+        :raise: ``UsageError`` indicating which provider configuration was
+                missing.
+        """
+        raise UsageError(
+            "Configuration file must include a "
+            "{!r} config stanza.".format(provider)
+        )
+
+    def _runner_VAGRANT(self, package_source,
+                        dataset_backend, provider_config):
+        """
+        :param PackageSource package_source: The source of omnibus packages.
+        :param DatasetBackend dataset_backend: A ``DatasetBackend`` constant.
+        :param provider_config: The ``vagrant`` section of the
+            testing configuration file.  Since the Vagrant runner accepts no
+            configuration, this is ignored.
+        :returns: ``VagrantRunner``
+        """
+        return VagrantRunner(
+            config=self['config'],
+            top_level=self.top_level,
+            distribution=self['distribution'],
+            package_source=package_source,
+            variants=self['variants'],
+            dataset_backend=dataset_backend,
+            dataset_backend_configuration=self.dataset_backend_configuration()
+        )
+
+    def _runner_MANAGED(self, package_source, dataset_backend,
+                        provider_config):
+        """
+        :param PackageSource package_source: The source of omnibus packages.
+        :param DatasetBackend dataset_backend: A ``DatasetBackend`` constant.
+        :param provider_config: The ``managed`` section of the
+            testing configuration file.  The section of the configuration
+            file should look something like:
+
+                managed:
+                  addresses:
+                    - "172.16.255.240"
+                    - "172.16.255.241"
+                  distribution: "centos-7"
+        :returns: ``ManagedRunner``.
+        """
+        if provider_config is None:
+            self._provider_config_missing("managed")
+
+        if not provider_config.get("upgrade"):
+            package_source = None
+
+        return ManagedRunner(
+            node_addresses=provider_config['addresses'],
+            package_source=package_source,
+            # TODO LATER Might be nice if this were part of
+            # provider_config. See FLOC-2078.
+            distribution=self['distribution'],
+            dataset_backend=dataset_backend,
+            dataset_backend_configuration=self.dataset_backend_configuration(),
+        )
+
+    def _libcloud_runner(self, package_source, dataset_backend,
+                         provider, provider_config):
+        """
+        Run some nodes using ``libcloud``.
+
+        By default, two nodes are run.  This can be overridden by setting
+        ``FLOCKER_ACCEPTANCE_NUM_NODES`` in the environment.
```

Comment:
This part of the docstring can be removed, as it no longer uses the environment variable.